Tools/Schema.NET.Tool/SourceUtility.cs

```diff
@@ -39,37 +46,31 @@ public static string RenderItems<T>(IEnumerable<T> items, Func<T, string> action
 #endif
 
         var stringBuilder = new StringBuilder();
-        foreach (var item in items)
-        {
-            stringBuilder.Append(action(item));
-        }
-
-        return stringBuilder.ToString();
-    }
-
-    public static string RenderItems<T>(IEnumerable<T> items, Func<T, int, string> action)
-    {
-#if NET6_0_OR_GREATER
-            ArgumentNullException.ThrowIfNull(items);
-            ArgumentNullException.ThrowIfNull(action);
-#else
-        if (items is null)
+        for (var i = 0; i < items.Count; ++i)
         {
-            throw new ArgumentNullException(nameof(items));
-        }
+            var item = items.ElementAt(i);
+            var line = action(i, indent, item).Replace(Environment.NewLine, Environment.NewLine + new string(Space, indent));
+            var isLast = i == items.Count - 1;
 
-        if (action is null)
-        {
-            throw new ArgumentNullException(nameof(action));
-        }
-#endif
+            stringBuilder.Append(Space, indent);
 
-        var stringBuilder = new StringBuilder();
-        var i = 0;
-        foreach (var item in items)
-        {
-            stringBuilder.Append(action(item, i));
-            i++;
+#pragma warning disable IDE0072 // Add missing cases
```

Comment:
I'm surprised we need to disable this as you've got the discard/catch-all rule - do you know what's up with that?

Reply:
I think this is a VS bug:

![image](https://user-images.githubusercontent.com/2270364/181774268-5c23e6a4-9531-4451-9463-b802a471883f.png)

I've noticed a lot of times in the past where I've had to ignore warnings only to find that the warning no longer occurs. This usually happens when a new .NET version is around the corner.